File: source/triangleNumbers/baseline.js
const REPEATS = 100;
const n = Number(process.argv[2]);

const now = Date.now();

let repeat = REPEATS;

while (repeat--) {
  let sum = 0;
  for (let i = 1; i <= n; i++) {
    sum += i;
  }

  if (repeat == 0) {
    console.log(`Result: ${sum}`);
  }
}

console.log(`Used time (average, in ms): ${(Date.now() - now) / REPEATS}`);

File: source/triangleNumbers/baseline.rs
use std::time::Instant;

pub fn triangle_number(n: u32) -> u64 {
    (1..=n as u64).fold(0, |sum, x| sum.wrapping_add(x))
}

const REPEAT: u128 = 1000;

fn main() {
    let args: Vec<String> = std::env::args().collect();

    let n: u32 = args[1].parse().unwrap();
    let now = Instant::now();

    let mut a = 0;

    for _ in 0..REPEAT {
        a = triangle_number(n)
    }

    println!(
        "Result: {},\nUsed time (average, in milliseconds): {}",
        a,
        (now.elapsed().as_nanos() / REPEAT) as f32 / 1_000_000.0
    );
}
